backend/tests/test.cpp

```diff
@@ -0,0 +1,29 @@
+#define CATCH_CONFIG_MAIN
+#include <regex>
+#include <catch2/catch_all.hpp>
+#include "../src/passwords.cpp"
+
+std::unordered_set<std::string> passwordSet = passwords::generatePasswords(5, 24);
+
+bool hasLettersAndDigit(std::string str)
+{
+    const std::regex pattern("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).+$");
+    return regex_match(str, pattern);
+}
+
+bool hasSymbol(std::string str)
+{
+    return str.find_first_not_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890_") != std::string::npos;
+}
+
+TEST_CASE("Test basic password requirements")
+{
+    // function generated 5 passwords
+    REQUIRE(passwordSet.size() == 5);
```

Comment:
**issue**: You don't want to use `REQUIRE` here because that stops the program from running the rest of the tests. Use `CHECK` instead which is the typical notion of assert in other languages. For more info, https://github.com/catchorg/Catch2/blob/devel/docs/assertions.md.